printf `%PRIu64` format string set incorrectly when cross-compiling to Windows with MinGW-w64

I am cross-compiling the pyinstaller bootloader from Linux (amd64) to Windows using mingw-w64, and then running pyinstaller inside wine to build Windows binaries on Linux.
This currently works with pyinstaller 4.2 (with the patch from https://github.com/pyinstaller/pyinstaller/issues/5544, which has since been merged into 4.4) - see [code here](https://github.com/spesmilo/electrum/blob/3e42f2f33d56308455a5c75a460f2262c7a576bd/contrib/build-wine/prepare-wine.sh#L101).
I am trying to update pyinstaller to 4.10 now. The bootloader builds successfully, and pyinstaller manages to create my binary, however the binary crashes at runtime. Details follow.

-----

Setting `console=True` and `debug=True` to the pyinstaller build, I ran the built binary
(tested both on Windows-10-21H1, and on wine):

```
[18192] LOADER: Running pyiboot01_bootstrap.py
Traceback (most recent call last):
  File "PyInstaller\loader\pyiboot01_bootstrap.py", line 20, in <module>
  File "PyInstaller\loader\pyimod03_importers.py", line 573, in install
  File "PyInstaller\loader\pyimod03_importers.py", line 128, in __init__
ImportError: Cannot load frozen modules.
[18192] Failed to execute script 'pyiboot01_bootstrap' due to unhandled exception!
```

https://github.com/pyinstaller/pyinstaller/blob/669313ba4c5c1403ebeb335c35cb68c8c6ba5dd4/PyInstaller/loader/pyimod03_importers.py#L128

Then, I patched `PyInstaller/loader/pyimod03_importers.py` around that line, and tried again:

```
PS C:\tmp> .\electrum-4.2.0-7-g1bb7ef92e-dirty.exe
[4936] PyInstaller Bootloader 3.x
[4936] LOADER: executable is C:\tmp\electrum-4.2.0-7-g1bb7ef92e-dirty.exe
[4936] LOADER: homepath is C:\tmp
[...]
[18192] LOADER: Installing PYZ archive with Python modules.
[18192] LOADER: PYZ archive: PYZ-00.pyz
[18192] LOADER: Running pyiboot01_bootstrap.py
Traceback (most recent call last):
  File "PyInstaller\loader\pyiboot01_bootstrap.py", line 20, in <module>
  File "PyInstaller\loader\pyimod03_importers.py", line 577, in install
  File "PyInstaller\loader\pyimod03_importers.py", line 132, in __init__
ImportError: Cannot load frozen modules. -- tried to load C:\Users\User\AppData\Local\Temp\_MEI49362\base_library.zip. got ArchiveReadError('C:\\Users\\User\\AppData\\Local\\Temp\\_MEI49362\\base_library.zip is not a valid ZlibArchiveReader archive file')
tried to load C:\Users\User\AppData\Local\Temp\_MEI49362\lib-dynload. got FileNotFoundError(2, 'No such file or directory')
tried to load C:\Users\User\AppData\Local\Temp\_MEI49362. got PermissionError(13, 'Permission denied')
tried to load C:\tmp\electrum-4.2.0-7-g1bb7ef92e-dirty.exe?%I64u. got OSError(22, 'Invalid argument')

[18192] Failed to execute script 'pyiboot01_bootstrap' due to unhandled exception!
[...]
```

`C:\tmp\electrum-4.2.0-7-g1bb7ef92e-dirty.exe?%I64u` -- that's a weird looking filename :)

https://github.com/pyinstaller/pyinstaller/blob/669313ba4c5c1403ebeb335c35cb68c8c6ba5dd4/bootloader/src/pyi_pythonlib.c#L623

So `%I64u` seems to be a C format string for `printf`, which did not get substituted.
That line was changed in https://github.com/pyinstaller/pyinstaller/pull/5610, first included in version 4.3.

-----

I have found a workaround: (although I guess this is not how it should be fixed)

```patch
diff --git a/bootloader/src/pyi_pythonlib.c b/bootloader/src/pyi_pythonlib.c
index 8bbf1d23..52ff7bc3 100644
--- a/bootloader/src/pyi_pythonlib.c
+++ b/bootloader/src/pyi_pythonlib.c
@@ -620,7 +620,7 @@ pyi_pylib_install_zlib(ARCHIVE_STATUS *status, TOC *ptoc)
      */
     archivename_obj = PI_PyUnicode_DecodeFSDefault(status->archivename);
 #endif
-    zlib_entry = PI_PyUnicode_FromFormat("%U?%" PRIu64, archivename_obj, zlibpos);
+    zlib_entry = PI_PyUnicode_FromFormat("%U?%llu", archivename_obj, zlibpos);
     PI_Py_DecRef(archivename_obj);
 
     sys_path = PI_PySys_GetObject("path");

```

-----

`PRIu64` is defined in the `inttypes.h` header.

I am building on Ubuntu 20.04 LTS, using `mingw-w64=7.0.0-2`.

See `inttypes.h` for that version here. Apparently the `else` branch must have been taken:
https://github.com/mirror/mingw-w64/blob/5eece15a2eb2e97da10d1e8cf5ff46e72b3bd768/mingw-w64-headers/crt/inttypes.h#L25-L46
(`#define PRIu64 "I64u"`)

Note that mingw-w64 8.0 includes a [change](https://github.com/mirror/mingw-w64/commit/bfd33f6c0ec5e652cc9911857dd1492ece8d8383) that I guess might set `__USE_MINGW_ANSI_STDIO` in this build configuration, which would mean that the first branch of the above `if` in `inttypes.h` gets taken.
(`#define PRIu64 "llu"`)


-----


<details>
 <summary>pyinstaller/bootloader/build/config.log</summary>

```
# project PyInstallerBootloader (nodist) configured on Fri Mar 18 20:14:50 2022 by
# waf 2.0.20 (abi 20, python 3080af0 on linux)
# using ./waf all CC=i686-w64-mingw32-gcc CFLAGS=-static                               -Wno-dangling-else                               -Wno-error=unused-value                               -Wno-error=implicit-function-declaration                               -Wno-error=int-to-pointer-cast                               -Wno-error=stringop-truncation
#
----------------------------------------
Setting top to
/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader
----------------------------------------
Setting out to
/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build
----------------------------------------
Python Version
3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
----------------------------------------
Checking for 'gcc' (C compiler)
find program=['gcc', 'cc'] paths=['/home/user/.local/bin', '/usr/local/sbin', '/usr/local/bin', '/usr/sbin', '/usr/bin', '/sbin', '/bin'] var='CC' -> ['i686-w64-mingw32-gcc']
['i686-w64-mingw32-gcc', '-dM', '-E', '-']
out: #define __DBL_MIN_EXP__ (-1021)
#define __FLT32X_MAX_EXP__ 1024
#define __pentiumpro__ 1
#define __UINT_LEAST16_MAX__ 0xffff
#define __ATOMIC_ACQUIRE 2
#define __FLT128_MAX_10_EXP__ 4932
#define __FLT_MIN__ 1.17549435082228750796873653722224568e-38F
#define __GCC_IEC_559_COMPLEX 2
#define __UINT_LEAST8_TYPE__ unsigned char
#define __SIZEOF_FLOAT80__ 12
#define _WIN32 1
#define __INTMAX_C(c) c ## LL
#define __CHAR_BIT__ 8
#define __UINT8_MAX__ 0xff
#define __WINT_MAX__ 0xffff
#define __FLT32_MIN_EXP__ (-125)
#define __ORDER_LITTLE_ENDIAN__ 1234
#define __SIZE_MAX__ 0xffffffffU
#define __WCHAR_MAX__ 0xffff
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_1 1
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_2 1
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_4 1
#define __DBL_DENORM_MIN__ ((double)4.94065645841246544176568792868221372e-324L)
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_8 1
#define __GCC_ATOMIC_CHAR_LOCK_FREE 2
#define __GCC_IEC_559 2
#define __FLT32X_DECIMAL_DIG__ 17
#define __FLT_EVAL_METHOD__ 2
#define __FLT64_DECIMAL_DIG__ 17
#define __GCC_ATOMIC_CHAR32_T_LOCK_FREE 2
#define __UINT_FAST64_MAX__ 0xffffffffffffffffULL
#define __SIG_ATOMIC_TYPE__ int
#define __DBL_MIN_10_EXP__ (-307)
#define __FINITE_MATH_ONLY__ 0
#define __GNUC_PATCHLEVEL__ 0
#define __FLT32_HAS_DENORM__ 1
#define __UINT_FAST8_MAX__ 0xff
#define __has_include(STR) __has_include__(STR)
#define _stdcall __attribute__((__stdcall__))
#define __DEC64_MAX_EXP__ 385
#define __INT8_C(c) c
#define __INT_LEAST8_WIDTH__ 8
#define __UINT_LEAST64_MAX__ 0xffffffffffffffffULL
#define __SHRT_MAX__ 0x7fff
#define __LDBL_MAX__ 1.18973149535723176502126385303097021e+4932L
#define __FLT64X_MAX_10_EXP__ 4932
#define __UINT_LEAST8_MAX__ 0xff
#define __GCC_ATOMIC_BOOL_LOCK_FREE 2
#define __FLT128_DENORM_MIN__ 6.47517511943802511092443895822764655e-4966F128
#define __UINTMAX_TYPE__ long long unsigned int
#define __DEC32_EPSILON__ 1E-6DF
#define __FLT_EVAL_METHOD_TS_18661_3__ 2
#define __UINT32_MAX__ 0xffffffffU
#define __LDBL_MAX_EXP__ 16384
#define __FLT128_MIN_EXP__ (-16381)
#define __WINT_MIN__ 0
#define __FLT128_MIN_10_EXP__ (-4931)
#define __INT_LEAST16_WIDTH__ 16
#define __SCHAR_MAX__ 0x7f
#define __FLT128_MANT_DIG__ 113
#define __WCHAR_MIN__ 0
#define __INT64_C(c) c ## LL
#define __DBL_DIG__ 15
#define __GCC_ATOMIC_POINTER_LOCK_FREE 2
#define __FLT64X_MANT_DIG__ 64
#define __SIZEOF_INT__ 4
#define __SIZEOF_POINTER__ 4
#define __USER_LABEL_PREFIX__ _
#define __FLT64X_EPSILON__ 1.08420217248550443400745280086994171e-19F64x
#define __STDC_HOSTED__ 1
#define __WIN32 1
#define __LDBL_HAS_INFINITY__ 1
#define __FLT32_DIG__ 6
#define __FLT_EPSILON__ 1.19209289550781250000000000000000000e-7F
#define __SHRT_WIDTH__ 16
#define __LDBL_MIN__ 3.36210314311209350626267781732175260e-4932L
#define __STDC_UTF_16__ 1
#define __DEC32_MAX__ 9.999999E96DF
#define __FLT64X_DENORM_MIN__ 3.64519953188247460252840593361941982e-4951F64x
#define __MINGW32__ 1
#define __FLT32X_HAS_INFINITY__ 1
#define __INT32_MAX__ 0x7fffffff
#define __INT_WIDTH__ 32
#define __SIZEOF_LONG__ 4
#define __UINT16_C(c) c
#define __PTRDIFF_WIDTH__ 32
#define __DECIMAL_DIG__ 21
#define __FLT64_EPSILON__ 2.22044604925031308084726333618164062e-16F64
#define __INTMAX_WIDTH__ 64
#define __has_include_next(STR) __has_include_next__(STR)
#define __FLT64X_MIN_10_EXP__ (-4931)
#define __LDBL_HAS_QUIET_NAN__ 1
#define __FLT64_MANT_DIG__ 53
#define __GNUC__ 9
#define _cdecl __attribute__((__cdecl__))
#define __FLT_HAS_DENORM__ 1
#define __SIZEOF_LONG_DOUBLE__ 12
#define __BIGGEST_ALIGNMENT__ 16
#define __FLT64_MAX_10_EXP__ 308
#define __i686 1
#define __DBL_MAX__ ((double)1.79769313486231570814527423731704357e+308L)
#define _thiscall __attribute__((__thiscall__))
#define __INT_FAST32_MAX__ 0x7fffffff
#define __WINNT 1
#define __DBL_HAS_INFINITY__ 1
#define __WINNT__ 1
#define __HAVE_SPECULATION_SAFE_VALUE 1
#define __DEC32_MIN_EXP__ (-94)
#define __INTPTR_WIDTH__ 32
#define __FLT32X_HAS_DENORM__ 1
#define __INT_FAST16_TYPE__ short int
#define _fastcall __attribute__((__fastcall__))
#define __LDBL_HAS_DENORM__ 1
#define __FLT128_HAS_INFINITY__ 1
#define __DEC128_MAX__ 9.999999999999999999999999999999999E6144DL
#define __INT_LEAST32_MAX__ 0x7fffffff
#define __USING_SJLJ_EXCEPTIONS__ 1
#define __DEC32_MIN__ 1E-95DF
#define __DBL_MAX_EXP__ 1024
#define __WCHAR_WIDTH__ 16
#define __FLT32_MAX__ 3.40282346638528859811704183484516925e+38F32
#define __DEC128_EPSILON__ 1E-33DL
#define __ATOMIC_HLE_RELEASE 131072
#define __WIN32__ 1
#define __PTRDIFF_MAX__ 0x7fffffff
#define __ATOMIC_HLE_ACQUIRE 65536
#define __FLT32_HAS_QUIET_NAN__ 1
#define __LONG_LONG_MAX__ 0x7fffffffffffffffLL
#define __SIZEOF_SIZE_T__ 4
#define __FLT64X_MIN_EXP__ (-16381)
#define __SIZEOF_WINT_T__ 2
#define __LONG_LONG_WIDTH__ 64
#define __FLT32_MAX_EXP__ 128
#define __GXX_ABI_VERSION 1013
#define __FLT_MIN_EXP__ (-125)
#define __i686__ 1
#define __FLT64X_HAS_QUIET_NAN__ 1
#define __INT_FAST64_TYPE__ long long int
#define __FLT64_DENORM_MIN__ 4.94065645841246544176568792868221372e-324F64
#define __DBL_MIN__ ((double)2.22507385850720138309023271733240406e-308L)
#define __FLT32X_EPSILON__ 2.22044604925031308084726333618164062e-16F32x
#define __DECIMAL_BID_FORMAT__ 1
#define __FLT64_MIN_EXP__ (-1021)
#define __GXX_TYPEINFO_EQUALITY_INLINE 0
#define __FLT64_MIN_10_EXP__ (-307)
#define __FLT64X_DECIMAL_DIG__ 21
#define __DEC128_MIN__ 1E-6143DL
#define __REGISTER_PREFIX__ 
#define __UINT16_MAX__ 0xffff
#define __DBL_HAS_DENORM__ 1
#define __cdecl __attribute__((__cdecl__))
#define __FLT32_MIN__ 1.17549435082228750796873653722224568e-38F32
#define __UINT8_TYPE__ unsigned char
#define __NO_INLINE__ 1
#define __i386 1
#define __FLT_MANT_DIG__ 24
#define __LDBL_DECIMAL_DIG__ 21
#define __VERSION__ "9.3-win32 20200320"
#define __UINT64_C(c) c ## ULL
#define __GCC_ATOMIC_INT_LOCK_FREE 2
#define __FLT128_MAX_EXP__ 16384
#define __FLT32_MANT_DIG__ 24
#define _X86_ 1
#define __FLOAT_WORD_ORDER__ __ORDER_LITTLE_ENDIAN__
#define __FLT128_HAS_DENORM__ 1
#define __FLT128_DIG__ 33
#define __SCHAR_WIDTH__ 8
#define __INT32_C(c) c
#define __DEC64_EPSILON__ 1E-15DD
#define __ORDER_PDP_ENDIAN__ 3412
#define __DEC128_MIN_EXP__ (-6142)
#define __code_model_32__ 1
#define __FLT32_MAX_10_EXP__ 38
#define __INT_FAST32_TYPE__ int
#define __UINT_LEAST16_TYPE__ short unsigned int
#define __FLT64X_HAS_INFINITY__ 1
#define __INT16_MAX__ 0x7fff
#define __i386__ 1
#define __SIZE_TYPE__ unsigned int
#define __UINT64_MAX__ 0xffffffffffffffffULL
#define __FLT64X_DIG__ 18
#define __INT8_TYPE__ signed char
#define __GCC_ASM_FLAG_OUTPUTS__ 1
#define __FLT_RADIX__ 2
#define __INT_LEAST16_TYPE__ short int
#define __LDBL_EPSILON__ 1.08420217248550443400745280086994171e-19L
#define __UINTMAX_C(c) c ## ULL
#define __SIG_ATOMIC_MAX__ 0x7fffffff
#define __GCC_ATOMIC_WCHAR_T_LOCK_FREE 2
#define __SIZEOF_PTRDIFF_T__ 4
#define __FLT32X_MANT_DIG__ 53
#define __FLT32X_MIN_EXP__ (-1021)
#define __DEC32_SUBNORMAL_MIN__ 0.000001E-95DF
#define __pentiumpro 1
#define __MSVCRT__ 1
#define __INT_FAST16_MAX__ 0x7fff
#define __FLT64_DIG__ 15
#define __UINT_FAST32_MAX__ 0xffffffffU
#define __UINT_LEAST64_TYPE__ long long unsigned int
#define __FLT_HAS_QUIET_NAN__ 1
#define __FLT_MAX_10_EXP__ 38
#define __LONG_MAX__ 0x7fffffffL
#define __FLT64X_HAS_DENORM__ 1
#define __DEC128_SUBNORMAL_MIN__ 0.000000000000000000000000000000001E-6143DL
#define __FLT_HAS_INFINITY__ 1
#define __UINT_FAST16_TYPE__ short unsigned int
#define __DEC64_MAX__ 9.999999999999999E384DD
#define __INT_FAST32_WIDTH__ 32
#define __CHAR16_TYPE__ short unsigned int
#define __PRAGMA_REDEFINE_EXTNAME 1
#define __SIZE_WIDTH__ 32
#define __SEG_FS 1
#define __INT_LEAST16_MAX__ 0x7fff
#define __DEC64_MANT_DIG__ 16
#define __INT64_MAX__ 0x7fffffffffffffffLL
#define __UINT_LEAST32_MAX__ 0xffffffffU
#define __SEG_GS 1
#define __FLT32_DENORM_MIN__ 1.40129846432481707092372958328991613e-45F32
#define __GCC_ATOMIC_LONG_LOCK_FREE 2
#define __SIG_ATOMIC_WIDTH__ 32
#define __INT_LEAST64_TYPE__ long long int
#define __INT16_TYPE__ short int
#define __INT_LEAST8_TYPE__ signed char
#define __STDC_VERSION__ 201710L
#define __DEC32_MAX_EXP__ 97
#define __INT_FAST8_MAX__ 0x7f
#define __FLT128_MAX__ 1.18973149535723176508575932662800702e+4932F128
#define __INTPTR_MAX__ 0x7fffffff
#define __GXX_MERGED_TYPEINFO_NAMES 0
#define __FLT64_HAS_QUIET_NAN__ 1
#define __stdcall __attribute__((__stdcall__))
#define __FLT32_MIN_10_EXP__ (-37)
#define __FLT32X_DIG__ 15
#define __LDBL_MANT_DIG__ 64
#define __DBL_HAS_QUIET_NAN__ 1
#define __FLT64_HAS_INFINITY__ 1
#define __FLT64X_MAX__ 1.18973149535723176502126385303097021e+4932F64x
#define __SIG_ATOMIC_MIN__ (-__SIG_ATOMIC_MAX__ - 1)
#define __INTPTR_TYPE__ int
#define __UINT16_TYPE__ short unsigned int
#define __WCHAR_TYPE__ short unsigned int
#define __SIZEOF_FLOAT__ 4
#define __UINTPTR_MAX__ 0xffffffffU
#define __INT_FAST64_WIDTH__ 64
#define __DEC64_MIN_EXP__ (-382)
#define __FLT32_DECIMAL_DIG__ 9
#define __INT_FAST64_MAX__ 0x7fffffffffffffffLL
#define __GCC_ATOMIC_TEST_AND_SET_TRUEVAL 1
#define __FLT_DIG__ 6
#define __FLT32_HAS_INFINITY__ 1
#define __FLT64X_MAX_EXP__ 16384
#define __UINT_FAST64_TYPE__ long long unsigned int
#define __INT_MAX__ 0x7fffffff
#define WIN32 1
#define __INT64_TYPE__ long long int
#define __FLT_MAX_EXP__ 128
#define __DBL_MANT_DIG__ 53
#define __SIZEOF_FLOAT128__ 16
#define __INT_LEAST64_MAX__ 0x7fffffffffffffffLL
#define __GCC_ATOMIC_CHAR16_T_LOCK_FREE 2
#define __DEC64_MIN__ 1E-383DD
#define __WINT_TYPE__ short unsigned int
#define __UINT_LEAST32_TYPE__ unsigned int
#define __SIZEOF_SHORT__ 2
#define __LDBL_MIN_EXP__ (-16381)
#define __FLT64_MAX__ 1.79769313486231570814527423731704357e+308F64
#define __WINT_WIDTH__ 16
#define __INT_LEAST8_MAX__ 0x7f
#define __FLT32X_MAX_10_EXP__ 308
#define __LDBL_MAX_10_EXP__ 4932
#define __ATOMIC_RELAXED 0
#define __DBL_EPSILON__ ((double)2.22044604925031308084726333618164062e-16L)
#define __thiscall __attribute__((__thiscall__))
#define __FLT128_MIN__ 3.36210314311209350626267781732175260e-4932F128
#define __UINT8_C(c) c
#define __FLT64_MAX_EXP__ 1024
#define __INT_LEAST32_TYPE__ int
#define __SIZEOF_WCHAR_T__ 2
#define __UINT64_TYPE__ long long unsigned int
#define __FLT128_HAS_QUIET_NAN__ 1
#define __INT_FAST8_TYPE__ signed char
#define __fastcall __attribute__((__fastcall__))
#define __FLT64X_MIN__ 3.36210314311209350626267781732175260e-4932F64x
#define __GNUC_STDC_INLINE__ 1
#define __FLT64_HAS_DENORM__ 1
#define __FLT32_EPSILON__ 1.19209289550781250000000000000000000e-7F32
#define __DBL_DECIMAL_DIG__ 17
#define __STDC_UTF_32__ 1
#define __INT_FAST8_WIDTH__ 8
#define __DEC_EVAL_METHOD__ 2
#define __FLT32X_MAX__ 1.79769313486231570814527423731704357e+308F32x
#define __ORDER_BIG_ENDIAN__ 4321
#define __UINT32_C(c) c ## U
#define __INTMAX_MAX__ 0x7fffffffffffffffLL
#define __BYTE_ORDER__ __ORDER_LITTLE_ENDIAN__
#define WINNT 1
#define __FLT_DENORM_MIN__ 1.40129846432481707092372958328991613e-45F
#define __INT8_MAX__ 0x7f
#define __LONG_WIDTH__ 32
#define __UINT_FAST32_TYPE__ unsigned int
#define __CHAR32_TYPE__ unsigned int
#define __FLT_MAX__ 3.40282346638528859811704183484516925e+38F
#define __INT32_TYPE__ int
#define __SIZEOF_DOUBLE__ 8
#define __FLT_MIN_10_EXP__ (-37)
#define __FLT64_MIN__ 2.22507385850720138309023271733240406e-308F64
#define __INT_LEAST32_WIDTH__ 32
#define __INTMAX_TYPE__ long long int
#define i386 1
#define _INTEGRAL_MAX_BITS 64
#define __DEC128_MAX_EXP__ 6145
#define __FLT32X_HAS_QUIET_NAN__ 1
#define __ATOMIC_CONSUME 1
#define __GNUC_MINOR__ 3
#define __INT_FAST16_WIDTH__ 16
#define __UINTMAX_MAX__ 0xffffffffffffffffULL
#define __DEC32_MANT_DIG__ 7
#define __FLT32X_DENORM_MIN__ 4.94065645841246544176568792868221372e-324F32x
#define __DBL_MAX_10_EXP__ 308
#define __LDBL_DENORM_MIN__ 3.64519953188247460252840593361941982e-4951L
#define __INT16_C(c) c
#define __STDC__ 1
#define __PTRDIFF_TYPE__ int
#define __ATOMIC_SEQ_CST 5
#define __UINT32_TYPE__ unsigned int
#define __FLT32X_MIN_10_EXP__ (-307)
#define __UINTPTR_TYPE__ unsigned int
#define __DEC64_SUBNORMAL_MIN__ 0.000000000000001E-383DD
#define __DEC128_MANT_DIG__ 34
#define __LDBL_MIN_10_EXP__ (-4931)
#define __FLT128_EPSILON__ 1.92592994438723585305597794258492732e-34F128
#define __SIZEOF_LONG_LONG__ 8
#define __FLT128_DECIMAL_DIG__ 36
#define __GCC_ATOMIC_LLONG_LOCK_FREE 2
#define __FLT32X_MIN__ 2.22507385850720138309023271733240406e-308F32x
#define __LDBL_DIG__ 18
#define __FLT_DECIMAL_DIG__ 9
#define __UINT_FAST16_MAX__ 0xffff
#define __GCC_ATOMIC_SHORT_LOCK_FREE 2
#define __INT_LEAST64_WIDTH__ 64
#define __UINT_FAST8_TYPE__ unsigned char
#define __ATOMIC_ACQ_REL 4
#define __ATOMIC_RELEASE 3
#define __declspec(x) __attribute__((x))

find program=['ar'] paths=['/home/user/.local/bin', '/usr/local/sbin', '/usr/local/bin', '/usr/sbin', '/usr/bin', '/sbin', '/bin'] var='AR' -> ['/usr/bin/ar']
i686-w64-mingw32-gcc
----------------------------------------
System
Assuming cross-compilation for Windows
----------------------------------------
Checking size of pointer
==>

        int main() {
            static int test_array[1 - 2 * !(sizeof(void *) == 4)];
            test_array[0] = 0;
            return 0;
        }
<==
[1/2] Compiling [32mbuild/.conf_check_643ebb47e5c1940fdd2fcedf4da4ce14/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_643ebb47e5c1940fdd2fcedf4da4ce14/testbuild/test.c.1.o']
[2/2] Linking [33mbuild/.conf_check_643ebb47e5c1940fdd2fcedf4da4ce14/testbuild/testprog.exe[0m

['i686-w64-mingw32-gcc', '-Wl,--enable-auto-import', 'test.c.1.o', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_643ebb47e5c1940fdd2fcedf4da4ce14/testbuild/testprog.exe', '-Wl,-Bstatic', '-Wl,-Bdynamic']
4
----------------------------------------
Platform
Windows-32bit detected based on compiler
----------------------------------------
Checking for compiler flags -m32
==>

int main(int argc, char **argv) {
	(void)argc; (void)argv;
	return 0;
}

<==
[1/1] Compiling [32mbuild/.conf_check_c20c3e81115158a949ca797b35b9c29e/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_c20c3e81115158a949ca797b35b9c29e/testbuild/test.c.1.o']
yes
----------------------------------------
Checking for linker flags -m32
==>

int main(int argc, char **argv) {
	(void)argc; (void)argv;
	return 0;
}

<==
[1/2] Compiling [32mbuild/.conf_check_9e1806166015a83897e6175b88f481fa/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_9e1806166015a83897e6175b88f481fa/testbuild/test.c.1.o']
[2/2] Linking [33mbuild/.conf_check_9e1806166015a83897e6175b88f481fa/testbuild/testprog.exe[0m

['i686-w64-mingw32-gcc', '-Wl,--enable-auto-import', '-m32', 'test.c.1.o', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_9e1806166015a83897e6175b88f481fa/testbuild/testprog.exe', '-Wl,-Bstatic', '-Wl,-Bdynamic']
yes
----------------------------------------
Checking for library user32
==>

int main(int argc, char **argv) {
	(void)argc; (void)argv;
	return 0;
}

<==
[1/2] Compiling [32mbuild/.conf_check_2765bb63d1a483a55fbabc75e7aa291c/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-Wno-unused-variable', '-Wno-unused-function', '-DWIN32', '-DNTDDI_VERSION=0x06010000', '-D_WIN32_WINNT=0x0601', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_2765bb63d1a483a55fbabc75e7aa291c/testbuild/test.c.1.o']
[2/2] Linking [33mbuild/.conf_check_2765bb63d1a483a55fbabc75e7aa291c/testbuild/testprog.exe[0m

['i686-w64-mingw32-gcc', '-Wl,--enable-auto-import', '-m32', '-Wl,--large-address-aware', 'test.c.1.o', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_2765bb63d1a483a55fbabc75e7aa291c/testbuild/testprog.exe', '-Wl,-Bstatic', '-Wl,-Bdynamic', '-luser32']
yes
----------------------------------------
Checking for library comctl32
==>

int main(int argc, char **argv) {
	(void)argc; (void)argv;
	return 0;
}

<==
[1/2] Compiling [32mbuild/.conf_check_dc39c627519909494f375200dc077663/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-Wno-unused-variable', '-Wno-unused-function', '-DWIN32', '-DNTDDI_VERSION=0x06010000', '-D_WIN32_WINNT=0x0601', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_dc39c627519909494f375200dc077663/testbuild/test.c.1.o']
[2/2] Linking [33mbuild/.conf_check_dc39c627519909494f375200dc077663/testbuild/testprog.exe[0m

['i686-w64-mingw32-gcc', '-Wl,--enable-auto-import', '-m32', '-Wl,--large-address-aware', 'test.c.1.o', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_dc39c627519909494f375200dc077663/testbuild/testprog.exe', '-Wl,-Bstatic', '-Wl,-Bdynamic', '-lcomctl32']
yes
----------------------------------------
Checking for library kernel32
==>

int main(int argc, char **argv) {
	(void)argc; (void)argv;
	return 0;
}

<==
[1/2] Compiling [32mbuild/.conf_check_e3adbd69b5d8fa18c9f3784dad5875f8/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-Wno-unused-variable', '-Wno-unused-function', '-DWIN32', '-DNTDDI_VERSION=0x06010000', '-D_WIN32_WINNT=0x0601', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_e3adbd69b5d8fa18c9f3784dad5875f8/testbuild/test.c.1.o']
[2/2] Linking [33mbuild/.conf_check_e3adbd69b5d8fa18c9f3784dad5875f8/testbuild/testprog.exe[0m

['i686-w64-mingw32-gcc', '-Wl,--enable-auto-import', '-m32', '-Wl,--large-address-aware', 'test.c.1.o', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_e3adbd69b5d8fa18c9f3784dad5875f8/testbuild/testprog.exe', '-Wl,-Bstatic', '-Wl,-Bdynamic', '-lkernel32']
yes
----------------------------------------
Checking for library advapi32
==>

int main(int argc, char **argv) {
	(void)argc; (void)argv;
	return 0;
}

<==
[1/2] Compiling [32mbuild/.conf_check_4412cc20769d057502f72ae44de9fe2c/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-Wno-unused-variable', '-Wno-unused-function', '-DWIN32', '-DNTDDI_VERSION=0x06010000', '-D_WIN32_WINNT=0x0601', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_4412cc20769d057502f72ae44de9fe2c/testbuild/test.c.1.o']
[2/2] Linking [33mbuild/.conf_check_4412cc20769d057502f72ae44de9fe2c/testbuild/testprog.exe[0m

['i686-w64-mingw32-gcc', '-Wl,--enable-auto-import', '-m32', '-Wl,--large-address-aware', 'test.c.1.o', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_4412cc20769d057502f72ae44de9fe2c/testbuild/testprog.exe', '-Wl,-Bstatic', '-Wl,-Bdynamic', '-ladvapi32']
yes
----------------------------------------
Checking for library gdi32
==>

int main(int argc, char **argv) {
	(void)argc; (void)argv;
	return 0;
}

<==
[1/2] Compiling [32mbuild/.conf_check_eaa24fd708ef5b641db2408eec3b1b75/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-Wno-unused-variable', '-Wno-unused-function', '-DWIN32', '-DNTDDI_VERSION=0x06010000', '-D_WIN32_WINNT=0x0601', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_eaa24fd708ef5b641db2408eec3b1b75/testbuild/test.c.1.o']
[2/2] Linking [33mbuild/.conf_check_eaa24fd708ef5b641db2408eec3b1b75/testbuild/testprog.exe[0m

['i686-w64-mingw32-gcc', '-Wl,--enable-auto-import', '-m32', '-Wl,--large-address-aware', 'test.c.1.o', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_eaa24fd708ef5b641db2408eec3b1b75/testbuild/testprog.exe', '-Wl,-Bstatic', '-Wl,-Bdynamic', '-lgdi32']
yes
----------------------------------------
Checking for library cmocka
==>

int main(int argc, char **argv) {
	(void)argc; (void)argv;
	return 0;
}

<==
[1/2] Compiling [32mbuild/.conf_check_58db325315388727d7b0ea1bc154bfae/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-Wno-unused-variable', '-Wno-unused-function', '-DWIN32', '-DNTDDI_VERSION=0x06010000', '-D_WIN32_WINNT=0x0601', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_58db325315388727d7b0ea1bc154bfae/testbuild/test.c.1.o']
[2/2] Linking [33mbuild/.conf_check_58db325315388727d7b0ea1bc154bfae/testbuild/testprog.exe[0m

['i686-w64-mingw32-gcc', '-Wl,--enable-auto-import', '-m32', '-Wl,--large-address-aware', 'test.c.1.o', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_58db325315388727d7b0ea1bc154bfae/testbuild/testprog.exe', '-Wl,-Bstatic', '-Wl,-Bdynamic', '-lcmocka']
err: /usr/bin/i686-w64-mingw32-ld: cannot find -lcmocka
collect2: error: ld returned 1 exit status

from /opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/tests: Test does not build: Traceback (most recent call last):
  File "/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/.waf3-2.0.20-a3a5949fb45bcd86794262f86d7977a7/waflib/Configure.py", line 335, in run_build
    bld.compile()
  File "/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/.waf3-2.0.20-a3a5949fb45bcd86794262f86d7977a7/waflib/Build.py", line 176, in compile
    raise Errors.BuildError(self.producer.error)
waflib.Errors.BuildError: Build failed
 -> task in 'testprog' failed with exit status 1 (run with -v to display more information)

not found
from /opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/tests: The configuration failed
----------------------------------------
Checking for function unsetenv
==>

    #include <stdlib.h>

    int main(int argc, char **argv) {
        void (*p)();

        (void)argc; (void)argv;
        p=(void(*)())(unsetenv);
        return !p;
    }

<==
[1/2] Compiling [32mbuild/.conf_check_2fb2d430badbe17c018cf04e913c0d28/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-Wno-unused-variable', '-Wno-unused-function', '-DWIN32', '-DNTDDI_VERSION=0x06010000', '-D_WIN32_WINNT=0x0601', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_2fb2d430badbe17c018cf04e913c0d28/testbuild/test.c.1.o']
err: ../test.c: In function ‘main’:
../test.c:8:23: error: ‘unsetenv’ undeclared (first use in this function)
    8 |         p=(void(*)())(unsetenv);
      |                       ^~~~~~~~
../test.c:8:23: note: each undeclared identifier is reported only once for each function it appears in

from /opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader: Test does not build: Traceback (most recent call last):
  File "/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/.waf3-2.0.20-a3a5949fb45bcd86794262f86d7977a7/waflib/Configure.py", line 335, in run_build
    bld.compile()
  File "/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/.waf3-2.0.20-a3a5949fb45bcd86794262f86d7977a7/waflib/Build.py", line 176, in compile
    raise Errors.BuildError(self.producer.error)
waflib.Errors.BuildError: Build failed
 -> task in 'testprog' failed with exit status 1 (run with -v to display more information)

no
from /opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader: The configuration failed
----------------------------------------
Checking for function mkdtemp
==>

    #include <stdlib.h>

    int main(int argc, char **argv) {
        void (*p)();

        (void)argc; (void)argv;
        p=(void(*)())(mkdtemp);
        return !p;
    }

<==
[1/2] Compiling [32mbuild/.conf_check_ce499de3920398c11a340f5198c2fee9/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-Wno-unused-variable', '-Wno-unused-function', '-DWIN32', '-DNTDDI_VERSION=0x06010000', '-D_WIN32_WINNT=0x0601', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_ce499de3920398c11a340f5198c2fee9/testbuild/test.c.1.o']
err: ../test.c: In function ‘main’:
../test.c:8:23: error: ‘mkdtemp’ undeclared (first use in this function); did you mean ‘mkstemp’?
    8 |         p=(void(*)())(mkdtemp);
      |                       ^~~~~~~
      |                       mkstemp
../test.c:8:23: note: each undeclared identifier is reported only once for each function it appears in

from /opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader: Test does not build: Traceback (most recent call last):
  File "/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/.waf3-2.0.20-a3a5949fb45bcd86794262f86d7977a7/waflib/Configure.py", line 335, in run_build
    bld.compile()
  File "/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/.waf3-2.0.20-a3a5949fb45bcd86794262f86d7977a7/waflib/Build.py", line 176, in compile
    raise Errors.BuildError(self.producer.error)
waflib.Errors.BuildError: Build failed
 -> task in 'testprog' failed with exit status 1 (run with -v to display more information)

no
from /opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader: The configuration failed
----------------------------------------
Checking for function dirname
==>

    #include <libgen.h>

    int main(int argc, char **argv) {
        void (*p)();

        (void)argc; (void)argv;
        p=(void(*)())(dirname);
        return !p;
    }

<==
[1/2] Compiling [32mbuild/.conf_check_d7b0c6ae8227ce3423b2af7279b8eb76/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-Wno-unused-variable', '-Wno-unused-function', '-DWIN32', '-DNTDDI_VERSION=0x06010000', '-D_WIN32_WINNT=0x0601', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_d7b0c6ae8227ce3423b2af7279b8eb76/testbuild/test.c.1.o']
[2/2] Linking [33mbuild/.conf_check_d7b0c6ae8227ce3423b2af7279b8eb76/testbuild/testprog.exe[0m

['i686-w64-mingw32-gcc', '-Wl,--enable-auto-import', '-m32', '-Wl,--large-address-aware', 'test.c.1.o', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_d7b0c6ae8227ce3423b2af7279b8eb76/testbuild/testprog.exe', '-Wl,-Bstatic', '-Wl,-Bdynamic']
yes
----------------------------------------
Checking for function basename
==>

    #include <libgen.h>

    int main(int argc, char **argv) {
        void (*p)();

        (void)argc; (void)argv;
        p=(void(*)())(basename);
        return !p;
    }

<==
[1/2] Compiling [32mbuild/.conf_check_a0286471fa1e961f2ee1e9a206c4b65e/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-Wno-unused-variable', '-Wno-unused-function', '-DWIN32', '-DNTDDI_VERSION=0x06010000', '-D_WIN32_WINNT=0x0601', '-DHAVE_DIRNAME=1', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_a0286471fa1e961f2ee1e9a206c4b65e/testbuild/test.c.1.o']
[2/2] Linking [33mbuild/.conf_check_a0286471fa1e961f2ee1e9a206c4b65e/testbuild/testprog.exe[0m

['i686-w64-mingw32-gcc', '-Wl,--enable-auto-import', '-m32', '-Wl,--large-address-aware', 'test.c.1.o', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_a0286471fa1e961f2ee1e9a206c4b65e/testbuild/testprog.exe', '-Wl,-Bstatic', '-Wl,-Bdynamic']
yes
----------------------------------------
Checking for function strndup
==>

    #include <string.h>

    int main(int argc, char **argv) {
        void (*p)();

        (void)argc; (void)argv;
        p=(void(*)())(strndup);
        return !p;
    }

<==
[1/2] Compiling [32mbuild/.conf_check_caffe7146e07123eb0f7c6dee4f27ab5/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-Wno-unused-variable', '-Wno-unused-function', '-DWIN32', '-DNTDDI_VERSION=0x06010000', '-D_WIN32_WINNT=0x0601', '-DHAVE_DIRNAME=1', '-DHAVE_BASENAME=1', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_caffe7146e07123eb0f7c6dee4f27ab5/testbuild/test.c.1.o']
err: ../test.c: In function ‘main’:
../test.c:8:23: error: ‘strndup’ undeclared (first use in this function); did you mean ‘strdup’?
    8 |         p=(void(*)())(strndup);
      |                       ^~~~~~~
      |                       strdup
../test.c:8:23: note: each undeclared identifier is reported only once for each function it appears in

from /opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader: Test does not build: Traceback (most recent call last):
  File "/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/.waf3-2.0.20-a3a5949fb45bcd86794262f86d7977a7/waflib/Configure.py", line 335, in run_build
    bld.compile()
  File "/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/.waf3-2.0.20-a3a5949fb45bcd86794262f86d7977a7/waflib/Build.py", line 176, in compile
    raise Errors.BuildError(self.producer.error)
waflib.Errors.BuildError: Build failed
 -> task in 'testprog' failed with exit status 1 (run with -v to display more information)

no
from /opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader: The configuration failed
----------------------------------------
Checking for function strnlen
==>

    #include <string.h>

    int main(int argc, char **argv) {
        void (*p)();

        (void)argc; (void)argv;
        p=(void(*)())(strnlen);
        return !p;
    }

<==
[1/2] Compiling [32mbuild/.conf_check_e7995e1a9c20af6741bff867b4d056e6/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '-O2', '-Wall', '-Werror', '-Wno-error=unused-variable', '-Wno-error=unused-function', '-Wno-unused-variable', '-Wno-unused-function', '-DWIN32', '-DNTDDI_VERSION=0x06010000', '-D_WIN32_WINNT=0x0601', '-DHAVE_DIRNAME=1', '-DHAVE_BASENAME=1', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_e7995e1a9c20af6741bff867b4d056e6/testbuild/test.c.1.o']
[2/2] Linking [33mbuild/.conf_check_e7995e1a9c20af6741bff867b4d056e6/testbuild/testprog.exe[0m

['i686-w64-mingw32-gcc', '-Wl,--enable-auto-import', '-m32', '-Wl,--large-address-aware', 'test.c.1.o', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_e7995e1a9c20af6741bff867b4d056e6/testbuild/testprog.exe', '-Wl,-Bstatic', '-Wl,-Bdynamic']
yes
----------------------------------------
Checking for program 'strip'
/usr/bin/strip
find program=['strip'] paths=['', '/home/user/.local/bin', '/usr/local/sbin', '/usr/local/bin', '/usr/sbin', '/usr/bin', '/sbin', '/bin'] var='STRIP' -> ['/usr/bin/strip']
----------------------------------------
Checking for program 'strip'
/usr/bin/strip
find program=['strip'] paths=['/home/user/.local/bin', '/usr/local/sbin', '/usr/local/bin', '/usr/sbin', '/usr/bin', '/sbin', '/bin'] var='STRIP' -> ['/usr/bin/strip']
```
</details>


-----

So... I am kind of at a loss here as to how this should be fixed... or if it is even an issue in pyinstaller.
Certainly it was frustrating to debug this - ideally at least there would be a compilation failure instead of a crash at runtime.

The workaround patch above works but makes the build fragile: if upstream pyinstaller introduces new usages of `PRIu64`, I will ~silently get new runtime bugs/crashes -- I could copy some of the `#define`s from `inttypes.h`.
Or maybe I could set `-D__USE_MINGW_ANSI_STDIO` explicitly until I can use newer mingw (until e.g. ubuntu 22.04) - though looking at the mailing list, [that seems discouraged](https://osdn.net/projects/mingw/lists/archive/users/2019-January/000201.html).

Maybe pyinstaller could be changed so that it either does not use `inttypes.h` or it explicitly patches parts of it?

Any thoughts? :)

--- a/bootloader/src/pyi_pythonlib.c
+++ b/bootloader/src/pyi_pythonlib.c
@@ -664,7 +664,7 @@ int
 pyi_pylib_install_zlib(ARCHIVE_STATUS *status, TOC *ptoc)
 {
     int rc = 0;
-    uint64_t zlibpos = status->pkgstart + ptoc->pos;
+    unsigned long long zlibpos = status->pkgstart + ptoc->pos;
     PyObject * sys_path, *zlib_entry, *archivename_obj;
 
     /* Note that sys.path contains PyUnicode on py3. Ensure
@@ -682,7 +682,7 @@ pyi_pylib_install_zlib(ARCHIVE_STATUS *status, TOC *ptoc)
      */
     archivename_obj = PI_PyUnicode_DecodeFSDefault(status->archivename);
 #endif
-    zlib_entry = PI_PyUnicode_FromFormat("%U?%" PRIu64, archivename_obj, zlibpos);
+    zlib_entry = PI_PyUnicode_FromFormat("%U?%llu", archivename_obj, zlibpos);
     PI_Py_DecRef(archivename_obj);
 
     sys_path = PI_PySys_GetObject("path");

--- a/news/6686.bugfix.rst
+++ b/news/6686.bugfix.rst
@@ -1,0 +1,5 @@
+Fix frozen module loading errors during the frozen application's bootstrap
+that arise from the bootloader's use of the ``PRIu64`` macro with the
+``PyUnicode_FromFormat`` call and the former translating to an incompatible
+format specifier on some toolchains (e.g., when cross-compiling Windows
+bootloader with mingw-w64 7.0.0 toolchain on Ubuntu 20.04 LTS).

